collector/collector.go

```diff
@@ -159,3 +185,28 @@ func (c *newRelicCollector) collectInstanceSummary(ch chan<- prometheus.Metric,
 		}
 	}
 }
+
+func (c *newRelicCollector) collectKeyTransactions(ch chan<- prometheus.Metric, wg *sync.WaitGroup) {
+	wg.Add(1)
+	defer wg.Done()
```

Comment:
@tmatias for the `wg.Add(1)` its make sense and resolve the point that you said above, when I wrote the code I forgot about this condition. Thanks!

I'll move the instruction before the `collectKeyTransactions` invocation, but for the `wg.Done()` it need to keep inside the function, because it can only be marked as done when the goroutine finish

Reply:
Or i could use a closure to do this, like:
```go
wg.Add(1)
go func() {
	defer wg.Done()
	c.collectKeyTransactions(ch)
}()
```
what do you think? It's seems unnecessary for me, but can be a approach to follow the pattern that you suggested

Reply:
@diogonicoleti Yeah, and it's also kinda ugly. 

:thinking: I see it working both ways, but maybe this uglier one is less error-prone.